### PR TITLE
feat: Implement DatetimeControl and add tests

### PR DESCRIPTION
**Pull request by TrangPham**

### CODE REVIEW

File: src/controls/DatetimeControl.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  handleChange(path: string, value: Date | null): void
+  path: string
+}
+
+const MyDatePicker = DatePicker.generatePicker<Date>(dateFnsGenerateConfig)
+
+export function DatetimeControl({ data, handleChange, path, label, visible, id }: DatetimeControlProps) {
+  const dateFormat = "MM/DD/YYYY"
+  return !visible ? null : (
+    <Form.Item label={label} id={id} name={path}>
```
Comment (NathanFarmer) [inline src/controls/DatetimeControl.tsx:17]: You can pass `required` here.

**Review by NathanFarmer — CHANGES_REQUESTED** [review]
Can you please add the `required` asterisk?

File: package.json [hunk 2]
```diff
@@ -52,6 +52,8 @@
     "@jsonforms/core": "^3.2.1",
     "@jsonforms/react": "^3.2.1",
     "antd": "^5.14.0",
+    "date-fns": "^3.3.1",
+    "rc-picker": "^4.2.0",
```
Comment (DrewHoo) [inline package.json:56]: these should be normal dependencies, right?

File: src/controls/DatetimeControl.tsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+export function DatetimeControl({
+  data,
+  handleChange,
+  path,
+  label,
+  visible,
+  id,
+  required,
+}: DatetimeControlProps) {
+  const dateFormat = "MM/DD/YYYY"
```
Comment (DrewHoo) [inline src/controls/DatetimeControl.tsx:23]: Can you make this configurable via UISchema or jsonschema / use browser-detected locale as a default?

**Review by DrewHoo — CHANGES_REQUESTED** [review]
G2G if you change peer dependencies to regular dependencies--locale configurability can wait TBH

**Comment by DrewHoo** [comment]
> @DrewHoo Looks like I need your review on it too since you previously requested changes.

Looks like dependencies still aren't updated? Also the DateControlOptions ought to be differentiated instead of added to a union with TextControlOptions. Something like this might work:
```
type JsonSchemaTypeToControlOptions<
  T,
> = T extends { type: infer U; format?: infer V }
? U extends "object" 
    ? unknown
    : U extends "string"
      ?  V extends "date" ? DateControlOptions : TextControlOptions
      : ...
```

**Comment by DrewHoo** [comment]
> @DrewHoo Is this no longer required? If not we can close this PR

Yeah I think so!

**Pull request closed without merging**